Arduino.mk

```
@@ -762,12 +762,12 @@ endif
 ########################################################################
 # Local sources
 
```

Excited to see this!

May I suggest:

```suggestion
SRC_DIR ?= $(dir $(realpath $(firstword $(MAKEFILE_LIST))))
```

This would default SRC_DIR to the location of the users Makefile. Since it would then always be defined, you could then print it in configuration vars without it being weird and use slash notation in your wildcard searches (if you cared to).